app/assets/javascripts/plot_positions.js

```
@@ -0,0 +1,32 @@
+var PlotPositions = (function() {
+  this.update_position = function(participant_incentive_id, participant_incentive_plot_position_id) {
+    $("#draggable-" + participant_incentive_id).draggable({
+      containment: $("#garden-individual"),
+      stop: function (event, ui) {
+        var end_position = $(this).position();
+        patch_to_server(participant_incentive_plot_position_id,
+          end_position.left / $("#garden-individual").width() * 100,
+          end_position.top / $("#garden-individual").height() * 100
+        );
+      }
+    })
+  };
+
+  this.patch_to_server = function(plot_position_id, x_coordinate_percentage, y_coordinate_percentage) {
+    $.ajax({
+      type: "PATCH",
+      url: "/plot_positions/" + plot_position_id,
+      data: {
+        "x": x_coordinate_percentage,
+        "y": y_coordinate_percentage
+      },
+      success: function (data, status) {
+        if ('success' === status) {
```

I see what you are saying, I'm going to vote for silent failures to update the garden image positions. This should only happen when the server is down, and if the server is down, then a page refresh would result in error as well. Would you have any recommendations on a better user experience?